misc/include/naive_rmm_tree.h

```diff
@@ -0,0 +1,274 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <cstdint>
+#include <limits>
+#include <algorithm>
+
+struct NaiveRmM
+{
+    using u64 = std::uint64_t;
```

Comment:
In general these kind of shortings doesn't really help you write code faster. Modern IDE's do a good job for covering these type of things. If you really want to use some kind of type shortcuts, they should be placed in a single place (something like types.h) and used everywhere.